WordPress/gutenberg
# Show uncategorized patterns on the Editor > Patterns page

## What?

Updates the Patterns page in the Site Editor to include uncategorized patterns.
1. Adds theme and plugin patterns that has no category registered to a "fake" category with the label 'Uncategorized".
2. Displays the uncategorized patterns on the page when that category is selected.

Closes https://github.com/WordPress/gutenberg/issues/52614

## Why?

Patterns without categories did not show on Editor > Patterns.
(Category is an optional parameter when registering patterns)

## How?


## Testing Instructions
Activate a block theme that includes patterns without a category, for example, Twenty Twenty-Three.
Go to Appearance > Editor > Patterns.
Confirm that the list of theme patterns has a category called Uncategorized.
Confirm that the number of items is correct
Confirm that the correct patterns are showing when the category is selected.

Confirm that the other categories and the search still work.
Confirm that uneditable patterns shows as locked, not synced.

## Screenshots or screencast 
https://github.com/WordPress/gutenberg/assets/7422055/83cb9ee7-0868-4144-9591-4d4c29277161
